### DISCUSSION TITLE
Connecting to Reolink E1 MQTT

### DISCUSSION THREAD

**Discussion by phatrickk** [QuantumEntangledAndy/neolink#110] [Q&A]
Hi,

Recently bought a Reolink E1 not realisng that it would not plug easily into home assistant straight away. Then found the orginal Neolink github and then subsequently this fork. Thanks for all your hardwork here! Hoping I can manage to get it into home assistant via this method, but I'm not real clever and I'm running into problems.

I installed gstreamer through chocolatey, and then I downloaded and unzipped the latest Win release of Neolink.

I created a "neolink.toml" file in that dir and this is my config:

```
bind = "0.0.0.0"

[[cameras]]
name = "Deck"
username = "****"
password = "****"
uid = "******"
  [cameras.mqtt]
  server = "192.168.0.137" # Address of the mqtt server
  port = 1883 # mqtt servers port
  credentials = ["****", "****"] # mqtt server login details
  	[cameras.mqtt.discovery]
  	topic = "homeassistant"
  	features = ["floodlight"]
``` 

When I then run the cmd - "./neolink mqtt --config=neolink.toml"

I get a fair bit of text come through with lots of green "info" (attached pic) and then an error:
ERROR: "Motion thread aborted: Communication error"
WARN: "Camera thread error: Communication error. Restarting

![Capture](https://github.com/QuantumEntangledAndy/neolink/assets/125029396/3ad32ea4-c4b0-41b5-9cc7-c44cd58e8518)

I had a quick look in my zigbee2mqtt install in homeassistant, and tried opening up joining to see if I could get anything discoverable, but otherwise I'm very confused and completely lost as to how to troubleshoot moving forward. Please help!

Thanks.

**Comment by QuantumEntangledAndy** [comment]
Have you been able to get the rtsp to connect?

I'll probably need you to test some things to troubleshoot this. For example if you remove the discovery things does it work?

**Reply by QuantumEntangledAndy** [reply]
Your error message means that we didn't understand the reply that the camera sent to us. Such as unexpected message format. Do you have motion disabled in the official client or something like that?

**Reply by phatrickk** [reply]
I don't think I have disabled motion... but also going through the app I can't seem find anything to enable or disable something that looks like motion?

I have no idea about rtsp. I've outlined the full process I've been through so far in the original post above. Basically installed gstreamer, installed neolink, created config, tried to execute it, error. Based on your question I'm guessing I'm missing something?

**Reply by QuantumEntangledAndy** [reply]
Ok let's try just the rtsp then. 

This is the command for rtsp

```bash
neolink rtsp --config "yourConfig.toml"
```

**Reply by phatrickk** [reply]
Seems to work? Action seems to stop at this point:
![Capture2](https://github.com/QuantumEntangledAndy/neolink/assets/125029396/9f15b7ec-3382-49df-b89d-94726f0217a5)

**Reply by QuantumEntangledAndy** [reply]
Ok no errors there then

Could you try this for your config

```toml
bind = "0.0.0.0"

[[cameras]]
name = "Deck"
username = "****"
password = "****"
uid = "******"
  [cameras.mqtt]
  server = "192.168.0.137"
  port = 1883
  credentials = ["****", "****"]
  enable_motion = false
```

It will turn off the motion so we can see if it's just that.

**Reply by QuantumEntangledAndy** [reply]
Btw what is your firmware version. I use E1 too (whole reason I started this thing is similar to your issue). Perhaps this is an updated firmware changing the message format.

**Reply by phatrickk** [reply]
Firmware is:
v3.0.0.115_20102200

Camera actually came with a newer firmware than this one (v3.0.0.4xxx I think), but this seems to be the latest available on the reolink download website. I 'upgraded' to see if that was the problem with connecting to the reolink integration.

Ok, with the new config, it now looks like this:
![Capture3](https://github.com/QuantumEntangledAndy/neolink/assets/125029396/b603a8af-e1a8-4349-a5e5-c3c42762097f)

I've just noticed that while nothing showed up in my Z2M section of home assistant, I now have a few entities assosciated with the camera in the home assistant entities list. The LED control works, but the camera is plugged in so battery doesn't (clever of me to add that one) and neither seemingly does the floodlight. But not sure if the E1 has one of those either.  When I click on the camera entity itself, I don't see anything, but not sure if I need to set up something in home assistant to get the feed working rather than just click on the entity?

![camera entity](https://github.com/QuantumEntangledAndy/neolink/assets/125029396/b32964d2-96dc-421f-a91b-eddd5955fe0e)

**Reply by QuantumEntangledAndy** [reply]
I see my E1 is on v2.0.0.587_1911080 so they might have changed something.

What's your current config look like. There's no floodlight or battery on the E1 so if they are there then remove them.

**Reply by phatrickk** [reply]
Yep, I updated the config to remove them. Now it looks like this:

```
bind = "0.0.0.0"

[[cameras]]
name = "Deck"
username = "****"
password = "****"
uid = "*******"
  [cameras.mqtt]
  server = "192.168.0.137"
  port = 1883 # mqtt servers port
  credentials = ["****", "****"]
  enable_motion = false
  	[cameras.mqtt.discovery]
  	topic = "homeassistant"
	features = ["led", "camera","ir", "motion", "reboot", "pt"]
```

I also updated firmware to v2.0.0.587_1911080. Tried the config again with motion enabled and ran into the same errors as before. Running again with motion = false works again, but still no feed in home assistant when I click the camera entity. Not sure if I need to do something else there.

**Reply by QuantumEntangledAndy** [reply]
What do you mean by "feed when you click on camera entity"

The mqtt command is control only (with a very basic image preview). It dosent have rtsp. That's what the rtsp command is for.
